Add filterBreadcrumbActions option

bjudson opened the pull request:
filterBreadcrumbActions is a function that returns `true` if an action should be added to the breadcrumbs.

The rationale for this is that some Redux apps use some actions internally to trigger state changes that are both prolific and not directly connected to user behavior. They can overwhelm the logs and have essentially no value for debugging purposes.

captbaritone commented:
How about instead of an array of types, we allow the user to supply a `filterBreadcrumbActions` function?

Also, I think the action should still be set as the `lastAction` even if it has been filtered out. Could you add a test to confirm that?

Thanks for the PR, I think this will be a great improvement.

bjudson commented:
@captbaritone Good idea. Updated PR with your suggestion. The test I added now confirms that the filtered action was omitted from breadcrumbs, but still included in lastAction. Thanks!

captbaritone commented on index.js:
```diff
@@ -21,11 +23,13 @@ function createRavenMiddleware(Raven, options = {}) {
-        message: action.type,
-        data: breadcrumbDataFromAction(action)
-      });
+      if (filterBreadcrumbActions(action) === true) {
```
I think it would be less confusing for this to just check for truthy values.

```
if (filterBreadcrumbActions(action)) {
```

bjudson replied:
Yeah, I did that at first, but thought being explicit might catch some bugs in the user-supplied function.

captbaritone replied:
I think truths/falsy is a perfectly valid return value from a filter predicate. Not nessesarily a bug.

captbaritone commented on README.md:
```diff
@@ -128,6 +128,16 @@ Each breadcrumb is assigned a category. By default all action breadcrumbs are
 given the category `"redux-action"`. If you would prefer a different category
 name, specify it here.
 
+#### `filterBreadcrumbActions` _(Function)_
```
This might also be worth adding to the "improvements" section above.

captbaritone commented on index.test.js:
```diff
@@ -178,4 +182,38 @@ describe("raven-for-redux", function() {
+        extra,
+        breadcrumbs
+      } = context.mockTransport.mock.calls[0][0].data;
+      // Even though the action isn't added to breadcrumbs, it should be sent with extra data
```
I think this would be better as a separate `it()` block, that way if it fails, we get a nice error message, but don't worry about it. I can always clean this up later.

captbaritone commented:
Looks good! I'll cut a point release.

bjudson commented:
Thanks for the quick feedback & merge! This is definitely going to improve our debugging process.

captbaritone commented:
Okay. Should be available in v1.1.0

Thanks again for your work on this!
